server/protocols/jmap/src/test/java/org/apache/james/jmap/JWTAuthenticationStrategyTest.java

```diff
@@ -0,0 +1,151 @@
+                .thenThrow(new MailboxException());
+
+        assertThatThrownBy(() -> testee.createMailboxSession(Stream.of(fakeAuthHeaderWithPrefix)))
+                .isExactlyInstanceOf(MailboxCreationException.class);
+    }
+
+    @Test
+    public void createMailboxSessionShouldReturnWhenAuthHeadersAreValid() throws Exception {
+        String username = "123456789";
+        String fakeAuthHeader = "blabla";
+        String fakeAuthHeaderWithPrefix = JWTAuthenticationStrategy.AUTHORIZATION_HEADER_PREFIX + fakeAuthHeader;
+        MailboxSession fakeMailboxSession = mock(MailboxSession.class);
+
+        when(stubTokenManager.verify(fakeAuthHeader)).thenReturn(true);
+        when(stubTokenManager.extractLogin(fakeAuthHeader)).thenReturn(username);
+        when(mockedMailboxManager.createSystemSession(eq(username), any(Logger.class)))
+                .thenReturn(fakeMailboxSession);
+
+        MailboxSession result = testee.createMailboxSession(Stream.of(fakeAuthHeaderWithPrefix));
+        assertThat(result).isEqualTo(fakeMailboxSession);
+    }
+
+    @Test
+    public void checkAuthorizationHeaderShouldReturnFalsewWhenAuthHeaderIsEmpty() {
+        assertThat(testee.checkAuthorizationHeader(Stream.empty())).isFalse();
+    }
+
+    @Test
+    public void checkAuthorizationHeaderShouldReturnFalsewWhenAuthHeaderIsInvalid() {
+        String fakeAuthHeader = "blabla";
+        String fakeAuthHeaderWithPrefix = JWTAuthenticationStrategy.AUTHORIZATION_HEADER_PREFIX + fakeAuthHeader;
+
+        when(stubTokenManager.verify(fakeAuthHeader)).thenReturn(false);
+
+        assertThat(testee.checkAuthorizationHeader(Stream.of(fakeAuthHeaderWithPrefix))).isFalse();
+    }
+
+    @Test
+    public void checkAuthorizationHeaderShouldReturnFalseWhenAuthHeadersAreInvalid() {
+        String fakeAuthHeader = "blabla";
+        String fakeAuthHeader2 = "blabla2";
+
+        when(stubTokenManager.verify(fakeAuthHeader)).thenReturn(false);
+        when(stubTokenManager.verify(fakeAuthHeader2)).thenReturn(false);
+
+        Stream<String> authHeadersStream = Stream.of(fakeAuthHeader, fakeAuthHeader2)
+                .map(h -> JWTAuthenticationStrategy.AUTHORIZATION_HEADER_PREFIX + h);
+        assertThat(testee.checkAuthorizationHeader(authHeadersStream)).isFalse();
+    }
+
+    @Test
+    public void checkAuthorizationHeaderShouldReturnTrueWhenAuthHeaderIsValid() {
+        String validAuthHeader = "blabla";
+        String validAuthHeaderWithPrefix = JWTAuthenticationStrategy.AUTHORIZATION_HEADER_PREFIX + validAuthHeader;
+
+        when(stubTokenManager.verify(validAuthHeader)).thenReturn(true);
+
+        assertThat(testee.checkAuthorizationHeader(Stream.of(validAuthHeaderWithPrefix))).isTrue();
+    }
+
+    @Test
+    public void checkAuthorizationHeaderShouldReturnTrueWhenOneAuthHeaderIsValid() {
+        String fakeAuthHeader = "blabla";
+        String validAuthHeader = "blabla2";
```

Comment:
Maybe instead of blabla and blabla2 you could have something more explicit, like fake and valid (in all this file).